File: _bench/fib/fib.nim
when defined(nimHasQuirky):
  {.push quirky: on.}

proc fib(n: uint64): uint64 =
  if n < 2:
    1
  else:
    fib(n - 1) + fib(n - 2)

when defined(nimHasQuirky):
  {.pop.}

let n: uint64 = 40
echo fib(n)

File: _bench/fib/fib.go
package main

import "fmt"

func fib(n uint64) uint64 {
	if n < 2 {
		return 1
	} else {
		return fib(n-1) + fib(n-2)
	}
}

func main() {
	var n uint64 = 40
	fmt.Println(fib(n))
}
